server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -73,6 +74,22 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
         return listBlobsByPrefix(null);
     }
 
+    @Override
+    public Map<String, BlobContainer> children() throws IOException {
+        // If we get duplicate files we should just take the last entry
+        Map<String, BlobContainer> builder = new HashMap<>();
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(path)) {
+            for (Path file : stream) {
+                final BasicFileAttributes attrs = Files.readAttributes(file, BasicFileAttributes.class);
+                if (attrs.isDirectory()) {
```

Comment:
why not just use `Files.isDirectory(file)`?

Reply:
jup ... this was just a left over from copying the logic of list by prefix, adjusting :)